model/src/main/kotlin/config/AdvisorConfiguration.kt

```diff
@@ -37,10 +37,15 @@ sealed class AdvisorConfiguration
  */
 data class BasicAuthConfiguration(
     /**
-     * The base URL of the provider.
+     * The URL of the server to make the requests against.
      */
     val serverUrl: String,
 
+    /**
+     * A URL to use as a base for browsing vulnerability details. Defaults to the server URL.
+     */
+    val browseUrl: String = serverUrl,
```

Comment:
IIUC the URL is unrelated to basic auth.
I'd like to suggest a different place but I have difficulties since I don't fully understand why the URL needs to be configurable. Can you give me a hint why it cannot be hard-coded in NexusIqService?

Reply:
It cannot be hard-coded because Nexus IQ is run on-premise, so each company running it uses a different host. Think of it like a GitHub Enterprise on-premise instance which has a company-specific REST API endpoint URL, and a company-specific URL to browse the source code.

Reply:
Would it make sense to drop the `BasicAuthConfiguration` class in a following change, to the following 
to remove the confusion of "how browseUrl relates to basic auth"?
```
data class AdvisorConfiguration(
   serverUrl: String, 
   browseUrl: String, authentications,
   username: String,
   password: String   
)
```

As soon as other AuthN mechanisms are added it might change to
```
data class AdvisorConfiguration(
   serverUrl: String, 
   browseUrl: String, authentications,
   authentication: Authentication
)

sealed class Authentication
```

Reply:
I believe we probably rather should just rename `BasicAuthConfiguration` to something else, as actually there is nothing basic-auth-specific in this class. Maybe just something generic like `RestApiConfiguration`? What you you think @MarcelBochtler?

Reply:
> as actually there is nothing basic-auth-specific in this class

I'd argue that username and password are basic-auth-specific. And configurations with e.g. a token based auth would be named differently.
IMO @fviernau's suggestion is the cleanest solution, and I'm also OK with dropping the `BasicAuthConfiguration` until other authentication methods need to be added for advisors.

Reply:
You also need to update `reference.conf` in this commit:
```
    nexusiq {
      serverUrl = "https://your-nexus-iq-server"
      username = username
      password = password
      browseUrl = "?"
    }
```

I'd move the properties to `AdvisorConfiguration` and remove `BasicAuthConfiguration`. Or if we want to support other auth methods in future the username+password should be moved to a separate class and also be nested in the conf like this:

```
    nexusiq {
      serverUrl = "https://your-nexus-iq-server"
      basicAuth {
        username = username
        password = password
      }
      browseUrl = "..."
    }
```

Reply:
I just had to look up how the configuration is implemented becuase I wondered where the `nexusiq` part is coming from. It seems to make the assumption that the configurations for different providers are similar which might be correct before, but now adding `browseUrl` adds this field also for other providers. Wouldn't it be better to have specific configuration classes for each provider?

Reply:
> I'd argue that username and password are basic-auth-specific.

To me, "basic (access) authentication" refers to the way *how* username and password are transmitted, i.e. via HTTP headers. A thing that has a `username` and `password` does not by itself make it a "basic authentication", as the credentials could be transmitted differently.

> Wouldn't it be better to have specific configuration classes for each provider?

That's indeed the key question, as having specific classes is also what we do in the `ScanStorageConfiguration` case. While I believe that even after the addition of `browseUrl` the current `BasicAuthConfiguration` class is so generic that it can probably also be used by other upcoming vulnerability providers, I'd be fine with renaming the class to `NexusIqConfiguration` to highlight the idea of having provider-specific configuration classes, but maybe rename the class again if we determine that all vulnerability providers would be using the same configuration class.